src/controllers/productController.ts

```diff
@@ -0,0 +1,118 @@
+import { Request, Response } from 'express'
+import cloudinary from 'cloudinary'
+import {
+  CLOUDINARY_CLOUD_NAME,
+  CLOUDINARY_API_KEY,
+  CLOUDINARY_API_SECRET,
+} from '../config/index'
+import Product from '../database/models/productModel'
+import Collection from '../database/models/collectionModel'
+
+cloudinary.v2.config({
+  cloud_name: CLOUDINARY_CLOUD_NAME,
+  api_key: CLOUDINARY_API_KEY,
+  api_secret: CLOUDINARY_API_SECRET,
+})
+
+/**
+ * Product Controller class
+ */
+export default class productController {
+  /**
+   * Adding product
+   * @param {Request} req - Express request object
+   * @param {Response} res - Express response object
+   * @returns {Promise<Response>} Promise that resolves to an Express response
+   */
+  static async addProduct(req: Request, res: Response): Promise<Response> {
+    try {
+      const {
+        name,
+        price,
+        category,
+        collectionId,
+        bonus,
+        sku,
+        quantity,
+        expiryDate,
+      } = req.body
+      if (
+        !name ||
+        !price ||
+        !category ||
+        !collectionId ||
+        !bonus ||
+        !sku ||
+        !quantity ||
+        !expiryDate
+      ) {
+        return res.status(400).json({ message: 'Bad request' })
+      }
```

Comment:
you are repeating some snippets of codes, I think must be done by joi schema  validation, You have passed out into your route.